### PR TITLE
Create the second step CS button option for linixUseCases

### PR DESCRIPTION
**Pull request by LinconOssa**
The second step of the assisted one is done

### CODE REVIEW

File: src/pages/privileges/outlets/linixUseCase/adding-linix-use-case/interface.tsx [hunk 1]
```diff
@@ -89,76 +102,79 @@ function AddingLinixUseCaseUI(props: AddingLinixUseCaseUIProps) {
...
-              buttonDescription={"Retorna a la página de inicio"}
-              route={"/privileges/linixUseCase"}
-            />
-          )}
+          <FormButtons
+            handleSubmit={isCurrentStateValid}
+            handleReset={isPreviousStepAvailable}
+            cancelButtonText="Atras"
+            submitButtonText="Siguiente"
+            disableReset={currentStep === 1} // Deshabilita el botón de reset
```
Comment (carin91) [inline src/pages/privileges/outlets/linixUseCase/adding-linix-use-case/interface.tsx:110]: Don't leave the code commented

File: src/pages/privileges/outlets/linixUseCase/adding-linix-use-case/forms/ClientServerButtonSelection/index.tsx [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+  withSubmitButtons?: boolean;
+  initialValues?: IClientServerButtonSelectionProps;
+  handleSubmit: (values: IClientServerButtonSelectionProps) => void;
+  onHasChanges?: (hasChanges: boolean) => void;
+  readOnly?: boolean;
+}
+
+function ClientServerButtonSelection(props: ClientServerButtonSelectionProps) {
+  const {
+    initialValues = {
```
Comment (carin91) [inline src/pages/privileges/outlets/linixUseCase/adding-linix-use-case/forms/ClientServerButtonSelection/index.tsx:29]: This must be refactor. why do you include a formik with these initial values?
initalValues = { csButtonOption: ""}

File: src/pages/privileges/outlets/linixUseCase/adding-linix-use-case/forms/ClientServerButtonSelection/index.tsx [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+import { useFormik } from "formik";
+
+import { EMessageType } from "@src/types/messages.types";
+import { IMessageState } from "@pages/privileges/outlets/users/types/forms.types";
+import { getData } from "@mocks/utils/dataMock.service";
+import { ClientServerButtonSelectionUI } from "./interface";
+
+const LOADING_TIMEOUT = 1500;
+
+export interface IClientServerButtonSelectionProps {
```
Comment (carin91) [inline src/pages/privileges/outlets/linixUseCase/adding-linix-use-case/forms/ClientServerButtonSelection/index.tsx:11]: Refactor this, you will need to know the value for csButton only. Do you need an interface for just (1) type? i would delete this interface

File: src/pages/privileges/outlets/linixUseCase/adding-linix-use-case/forms/ClientServerButtonSelection/index.tsx [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+  caseUseLinixName: string;
+  description: string;
+  webOptions: string;
+  ClientServerOption: string;
+  actionCaseUse: string;
+}
+
+interface ClientServerButtonSelectionProps {
+  withSubmitButtons?: boolean;
```
Comment (carin91) [inline src/pages/privileges/outlets/linixUseCase/adding-linix-use-case/forms/ClientServerButtonSelection/index.tsx:20]: Delete this, this is not necessary.

File: src/pages/privileges/outlets/linixUseCase/adding-linix-use-case/forms/ClientServerButtonSelection/index.tsx [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+function ClientServerButtonSelection(props: ClientServerButtonSelectionProps) {
+  const {
+    initialValues = {
+      caseUseLinixName: "",
+      description: "",
+      actionCaseUse: "",
+      webOptions: "",
+      ClientServerOption: "",
+    },
+    withSubmitButtons,
```
Comment (carin91) [inline src/pages/privileges/outlets/linixUseCase/adding-linix-use-case/forms/ClientServerButtonSelection/index.tsx:36]: Why is this necessary? delete this input and refactor the component

File: src/pages/privileges/outlets/linixUseCase/adding-linix-use-case/forms/ClientServerButtonSelection/index.tsx [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+  } = props;
+
+  const [loading, setLoading] = useState(false);
+  const [showMessage, setShowMessage] = useState<IMessageState>({
+    visible: false,
+  });
+  const [buttonOptionsMock, setbuttonOptionsMock] = useState<
+    Record<string, unknown>[]
+  >([]);
+  const [webOptions, setWebOptions] = useState<Record<string, unknown>[]>([]);
```
Comment (carin91) [inline src/pages/privileges/outlets/linixUseCase/adding-linix-use-case/forms/ClientServerButtonSelection/index.tsx:49]: Delete this, this component does not need to know about webOptions. Add to it, you need to consider where to locate every DB consumption. the webOptions should be consumed by the steps 1 and 5, so, for the activity of step 5/7 this "web-options" consumption must be relocate to the assisted form "add-linix-use-case"

For this activity, delete this state and its consumption.

File: src/pages/privileges/outlets/linixUseCase/adding-linix-use-case/forms/ClientServerButtonSelection/interface.tsx [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+    hasChanges,
+    formInvalid,
+    handleSubmitForm,
+    handleChangeForm,
+    readOnly,
+    buttonOptionsMock,
+    webOptions,
+  } = props;
+
+  if (withSubmitButtons) {
```
Comment (carin91) [inline src/pages/privileges/outlets/linixUseCase/adding-linix-use-case/forms/ClientServerButtonSelection/interface.tsx:128]: Avoid this, there must be just 1 return.
return(
{withSubmitButtons? <formButtons></formButtons>: <>{Render...}</>}
)

File: src/pages/privileges/outlets/linixUseCase/adding-linix-use-case/forms/ClientServerButtonSelection/interface.tsx [hunk 8]
```diff
@@ -0,0 +1,166 @@
...
+      <Stack
+        direction="column"
+        justifyContent="space-between"
+        padding="s050 s0"
+      >
+        <StyledSelectContainer>
+          <Select
+            label="Opción botón cliente servidor"
+            placeholder="Seleccione una opción"
+            name="Acción Caso de Uso"
```
Comment (carin91) [inline src/pages/privileges/outlets/linixUseCase/adding-linix-use-case/forms/ClientServerButtonSelection/interface.tsx:73]: Assign the same name as the formik name of this input, csButtonOption

File: src/pages/privileges/outlets/linixUseCase/adding-linix-use-case/forms/ClientServerButtonSelection/interface.tsx [hunk 9]
```diff
@@ -0,0 +1,166 @@
...
+            id="Acción Caso de Uso"
+            value={formik.values.actionCaseUse}
+            type="actionCaseUse"
```
Comment (carin91) [inline src/pages/privileges/outlets/linixUseCase/adding-linix-use-case/forms/ClientServerButtonSelection/interface.tsx:76]: Adjust this name, to the csButtonOption

**Review by carin91 — CHANGES_REQUESTED** [review]

File: src/components/forms/submit/FormButtons/index.tsx [hunk 10]
```diff
@@ -2,29 +2,40 @@ import { Button, Stack, inube } from "@inube/design-system";
...
+    disabledButtons,
+    loading,
+    cancelButtonText = "Cancelar",
+    submitButtonText = "Guardar",
+  } = props;
 
   return (
     <Stack direction="column" gap={inube.spacing.s300}>
-      <Stack direction="column">{children}</Stack>
+      {children && <Stack direction="column">{children}</Stack>}
```
[the remark this answers is not in the document]
Comment (LinconOssa) [reply src/components/forms/submit/FormButtons/index.tsx]: Ready

File: src/components/forms/submit/FormButtons/index.tsx [hunk 11]
```diff
@@ -2,29 +2,40 @@ import { Button, Stack, inube } from "@inube/design-system";
...
       <Stack justifyContent="flex-end" gap={inube.spacing.s100}>
         <ThemeProvider theme={inube}>
           <Button
             appearance="gray"
-            disabled={disabledButtons}
+            disabled={disableReset || disabledButtons} // Aquí se aplica la deshabilitación del botón de reset
```
[the remark this answers is not in the document]
Comment (LinconOssa) [reply src/components/forms/submit/FormButtons/index.tsx]: Ready

[the remark this answers is not in the document]
Comment (LinconOssa) [reply src/components/forms/submit/FormButtons/index.tsx]: Ready

File: src/components/forms/submit/FormButtons/index.tsx [hunk 12]
```diff
@@ -2,29 +2,40 @@ import { Button, Stack, inube } from "@inube/design-system";
 import { ThemeProvider } from "styled-components";
 
 interface FormButtonsProps {
-  children: React.ReactNode;
+  children?: React.ReactNode;
   handleSubmit: () => void;
   handleReset: () => void;
+  disableReset?: boolean; // Nuevo prop para deshabilitar el botón de reset
```
[the remark this answers is not in the document]
Comment (LinconOssa) [reply src/components/forms/submit/FormButtons/index.tsx]: Ready

File: src/pages/privileges/outlets/linixUseCase/adding-linix-use-case/forms/GeneralInformationForm/interface.tsx [hunk 13]
```diff
@@ -72,8 +72,11 @@ function RenderFormFields(
           type="text"
           size="compact"
           fullwidth
-          onChange={handleChangeForm}
+          onChange={(event: { target: { value: string } }) => {
```
[the remark this answers is not in the document]
Comment (LinconOssa) [reply src/pages/privileges/outlets/linixUseCase/adding-linix-use-case/forms/GeneralInformationForm/interface.tsx]: Ready good shape

File: src/pages/privileges/outlets/linixUseCase/adding-linix-use-case/interface.tsx (hunk 1 above)
Comment (LinconOssa) [reply src/pages/privileges/outlets/linixUseCase/adding-linix-use-case/interface.tsx]: Ready

File: src/pages/privileges/outlets/linixUseCase/adding-linix-use-case/forms/ClientServerButtonSelection/index.tsx (hunk 2 above)
Comment (LinconOssa) [reply src/pages/privileges/outlets/linixUseCase/adding-linix-use-case/forms/ClientServerButtonSelection/index.tsx]: Ready

File: src/pages/privileges/outlets/linixUseCase/adding-linix-use-case/forms/ClientServerButtonSelection/index.tsx (hunk 3 above)
Comment (LinconOssa) [reply src/pages/privileges/outlets/linixUseCase/adding-linix-use-case/forms/ClientServerButtonSelection/index.tsx]: Ready

File: src/pages/privileges/outlets/linixUseCase/adding-linix-use-case/forms/ClientServerButtonSelection/index.tsx (hunk 4 above)
Comment (LinconOssa) [reply src/pages/privileges/outlets/linixUseCase/adding-linix-use-case/forms/ClientServerButtonSelection/index.tsx]: Ready

File: src/pages/privileges/outlets/linixUseCase/adding-linix-use-case/forms/ClientServerButtonSelection/index.tsx (hunk 5 above)
Comment (LinconOssa) [reply src/pages/privileges/outlets/linixUseCase/adding-linix-use-case/forms/ClientServerButtonSelection/index.tsx]: Ready

File: src/pages/privileges/outlets/linixUseCase/adding-linix-use-case/forms/ClientServerButtonSelection/index.tsx (hunk 6 above)
Comment (LinconOssa) [reply src/pages/privileges/outlets/linixUseCase/adding-linix-use-case/forms/ClientServerButtonSelection/index.tsx]: Ready removed

File: src/pages/privileges/outlets/linixUseCase/adding-linix-use-case/forms/ClientServerButtonSelection/interface.tsx (hunk 7 above)
Comment (LinconOssa) [reply src/pages/privileges/outlets/linixUseCase/adding-linix-use-case/forms/ClientServerButtonSelection/interface.tsx]: ready

File: src/pages/privileges/outlets/linixUseCase/adding-linix-use-case/forms/ClientServerButtonSelection/interface.tsx (hunk 8 above)
Comment (LinconOssa) [reply src/pages/privileges/outlets/linixUseCase/adding-linix-use-case/forms/ClientServerButtonSelection/interface.tsx]: Ready and I think this is the same as the figma
image
image

File: src/pages/privileges/outlets/linixUseCase/adding-linix-use-case/forms/ClientServerButtonSelection/interface.tsx (hunk 9 above)
Comment (LinconOssa) [reply src/pages/privileges/outlets/linixUseCase/adding-linix-use-case/forms/ClientServerButtonSelection/interface.tsx]: Ready

**Review by carin91 — APPROVED** [review]

**Pull request merged**